packages/cli/src/parallel-tests/parallel-tests.handler.ts

```diff
@@ -1,48 +1,26 @@
-import { fork } from "child_process";
 import { cpus } from "os";
-import { join } from "path";
 import { TestCase } from "@alwaysmeticulous/api";
-import {
-  defer,
-  getMeticulousLocalDataDir,
-  METICULOUS_LOGGER_NAME,
-  ReplayEventsDependencies,
-  ReplayExecutionOptions,
-} from "@alwaysmeticulous/common";
+import { defer, METICULOUS_LOGGER_NAME } from "@alwaysmeticulous/common";
 import log from "loglevel";
-import { TestRun } from "../api/test-run.api";
-import { ScreenshotAssertionsEnabledOptions } from "../command-utils/common-types";
-import {
-  DetailedTestCaseResult,
-  MeticulousCliConfig,
-} from "../config/config.types";
-import { getReplayTargetForTestCase } from "../utils/config.utils";
-import { sortResults } from "../utils/run-all-tests.utils";
+import { DetailedTestCaseResult } from "../config/config.types";
 import { mergeResults } from "./merge-test-results";
-import { InitMessage, ResultMessage } from "./messages.types";
 import { TestRunProgress } from "./run-all-tests.types";
 
 export interface RunAllTestsInParallelOptions {
-  config: MeticulousCliConfig;
-  testRun: TestRun;
   testsToRun: TestCase[];
-  executionOptions: ReplayExecutionOptions;
-  screenshottingOptions: ScreenshotAssertionsEnabledOptions;
-  apiToken: string | null;
-  commitSha: string;
-
-  appUrl: string | null;
-  useAssetsSnapshottedInBaseSimulation: boolean;
   parallelTasks: number | null;
-  deflake: boolean;
-  replayEventsDependencies: ReplayEventsDependencies;
-
   maxRetriesOnFailure: number;
 
+  executeTest: (
```

Comment:
Refactored this to just take an executeTest method. This allows the complex logic here to be unit tested without launching new node processes from the tests, mock out all the ReplayExecutionOptions etc. It also provides a nicer seperation of concerns -- splitting out the logic that deals with launching child processes and communicating with them, vs the logic that chooses when to execute new tests, how many to execute in parallel at a time, how to auto-retry on failure etc.